src/test/netbase_tests.cpp

```diff
@@ -45,6 +45,30 @@ BOOST_AUTO_TEST_CASE(netbase_properties)
     BOOST_CHECK(CNetAddr("8.8.8.8").IsRoutable());
     BOOST_CHECK(CNetAddr("2001::1").IsRoutable());
     BOOST_CHECK(CNetAddr("127.0.0.1").IsValid());
+    BOOST_CHECK(CNetAddr("1.1.1").ToString() == "1.1.0.1");
+    BOOST_CHECK(CNetAddr("1.1").ToString() == "1.0.0.1");
+    BOOST_CHECK(CNetAddr("1111").ToString() == "0.0.4.87");
+    BOOST_CHECK(CNetAddr("1111111111").ToString() == "66.58.53.199");
+
+    // CNetAddr does not support ports making it dangerous to treat like CService
+    BOOST_CHECK(CNetAddr("250.2.2.2:7777").ToString() != CNetAddr("250.2.2.2", 7777).ToString());
+    BOOST_CHECK(CNetAddr("250.2.2.2:7777").ToString() == "::");
```

Comment:
I don't think it should be set in stone and in fact I want to change it, but having the tests confirm the behavior prior to making the change is valuable. If such a view is not-consistent with the Bitcoin testing philosophy I'm happy to remove these values.

There is an additional reason for these tests beyond just documenting undesirable behavior. Bitcoin uses different IP address parsing libraries for different platforms such as linux and windows. We want to avoid a case in which a change to one library results in differing results between platforms (maybe 1111 -> 1.1.0.1 or 1.0.1.1). These tests ensure that, at least for some of these strange values, we would catch such behavior.

>Edit: there is an IsValid on CNetAddr. This would need to return false in these cases, instead of valid nonsense.

I agree and that is a good point. If you don't mind I might take a look at fixing this behavior over the weekend.